### PR TITLE
supporting for multi repo

### PR DESCRIPTION
**Pull request by andrii-codefresh**

### CODE REVIEW

**Review by denis-codefresh — APPROVED** [review]

File: lib/EntrypointScriptBuilder.py [hunk 1]
```diff
@@ -175,20 +201,19 @@ def __init__(self, env):
...
 
-        helm_repo_username = env.get('HELMREPO_USERNAME')
-        helm_repo_password = env.get('HELMREPO_PASSWORD')
         for key, val in sorted(env.items()):
-            key_upper = key.upper()
-            if key_upper.startswith('CF_CTX_') and key_upper.endswith('_URL'):
-                repo_name = key_upper.replace('CF_CTX_', '', 1).replace('_URL', '', 1).replace('_', '-').lower()
-                repo_url = val
+            if key.startswith('CF_CTX_') and key.endswith('_URL'):
+                repo_name = key[7:len(key) - 4]
```
Comment (pasha-codefresh) [inline lib/EntrypointScriptBuilder.py:206]: add example

File: lib/EntrypointScriptBuilder.py [hunk 2]
```diff
@@ -61,14 +64,16 @@ def __init__(self, env):
...
+        self._extract_helm_repos(env)
+
         if self.helm_repository_context:
             context_integration = self._get_variables_from_helm_repo_integration(self.helm_repository_context)
             repo_url = context_integration.get('repositoryUrl')
             integration_name = context_integration.get('name')
             print("Using helm repo integration %s with URL %s" % (integration_name, repo_url))
             if repo_url is not None and integration_name is not None:
                 env['CF_CTX_'+integration_name+'_URL'] = repo_url
+                self.chart_repo_url = repo_url
```
Comment (pasha-codefresh) [inline lib/EntrypointScriptBuilder.py:76]: check if needed

File: lib/EntrypointScriptBuilder.py [hunk 3]
```diff
@@ -309,24 +328,30 @@ def _resolve_chart(env):
...
-            if self.credentials_in_arguments and (self.helm_repo_username is not None) and (
-                    self.helm_repo_password is not None):
-                helm_repo_add_cmd = 'helm repo add %s %s --username %s --password %s ' % (
-                repo_name, repo_url, self.helm_repo_username, self.helm_repo_password)
-            else:
-                helm_repo_add_cmd = 'helm repo add %s %s' % (repo_name, repo_url)
-            if self.dry_run:
-                helm_repo_add_cmd = 'echo ' + helm_repo_add_cmd
-            lines.append(helm_repo_add_cmd)
+        if (self.use_repos_for_auth_action.upper() == 'TRUE') or (self.action != 'auth'):
```
Comment (pasha-codefresh) [inline lib/EntrypointScriptBuilder.py:336]: convert to boolean before

File: lib/EntrypointScriptBuilder.py [hunk 4]
```diff
@@ -309,24 +328,30 @@ def _resolve_chart(env):
...
+            for repo_name, repo_url in sorted(self.helm_repos.items()):
+                helm_repo_username = os.getenv('CF_CTX_' + repo_name.upper() + '_HELMREPO_USERNAME')
```
Comment (pasha-codefresh) [inline lib/EntrypointScriptBuilder.py:338]: reuse get credentials ffunction

**Review by pasha-codefresh — CHANGES_REQUESTED** [review]

File: lib/EntrypointScriptBuilder.py (hunk 2 above)
Comment (andrii-codefresh) [reply lib/EntrypointScriptBuilder.py:76]: checked. it is OK

**Review by pasha-codefresh — APPROVED** [review]

**Pull request merged**